src/iterative_sorting/iterative_sorting.py

```diff
@@ -2,25 +2,52 @@
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
+        print(f"LOOP {i}")
+        print(f"{arr}")
         cur_index = i
         smallest_index = cur_index
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc) 
-             
-
+        for j in range(i, len(arr)):
+        # from the i'th position to the end of the array if j is smaller than the current value then j is the current smallest index
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
+        print(f"SMALLEST ELEMENT: {arr[smallest_index]}")
 
 
         # TO-DO: swap
-
+        temp = arr[i]
+        arr[i] = arr[smallest_index]
+        arr[smallest_index] = temp
 
 
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
-
+# def bubble_sort( arr ):
+#     for n in range(0,len(arr)-1):
+#       if arr[n] > arr[n+1]:
+#         unsorted = True
+#         arr[n], arr[n+1] = arr[n+1], arr[n]
+#       if n == len(arr-1) and unsorted:
+#         unsorted = False
+#         n = 0
+def bubble_sort(arr):
```

Comment:
This sorts properly, but it's not actually the proper implementation of bubble sort! The implementation you started above using a boolean as a flag is closer to the proper implementation. Let me know if you want to discuss it. In your free time, I suggest you look up a visualization of the algorithm (not solution), to understand what it should be doing.